qt/scientific_interfaces/ISISReflectometry/test/Experiment/ExperimentPresenterTest.h

```diff
@@ -820,13 +820,17 @@ class ExperimentPresenterTest : public CxxTest::TestSuite {
     previewRow.setProcessingInstructions(ROIType::Transmission, "12"s);
     previewRow.setTheta(2.3);
 
+    EXPECT_CALL(m_mainPresenter, notifySettingsChanged()).Times(1);
+
     presenter.notifyPreviewApplyRequested(previewRow);
     // Row with angle 2.3 is the last row in the look-up table
     auto row = presenter.experiment().lookupTableRows().back();
     TS_ASSERT_EQUALS(row.roiDetectorIDs().get(), "9");
     TS_ASSERT_EQUALS(row.processingInstructions().get(), "10");
     TS_ASSERT_EQUALS(row.backgroundProcessingInstructions().get(), "11");
     TS_ASSERT_EQUALS(row.transmissionProcessingInstructions().get(), "12");
+
+    verifyAndClear();
```

Comment:
Is `verifyAndClear` used at the end of each of the tests? If so, maybe it could go in the tearDown method?

Reply:
Good spot, I think it is now. I _think_ it's only technically needed if you have set expectations on the presenter or view (which all tests now do). I don't know if it would be an issue if you called `verifyAndClear` without any expectations set. I'm afraid I'm a bit pushed for time at the moment, so would you be happy if I open a maintenance issue for this one to look at during the release period?

Reply:
Yes doing it as part of maintenance is no problem. Which tests do you not see it in which are causing confusion?

Reply:
Great, thanks. Sorry, my comment wasn't very clear, I mean if for example you look at a totally different file such as PreviewPresenterTest.h I can't immediately see something that looks equivalent to what the `verifyAndClear` method seems to be doing, so I was keen to take the time to understand why that is (or spot where it is in PreviewPresenterTest.h).

Reply:
Ah I see, I'm not sure either - if you find out then let me know!

Reply:
This is because we have mocks like `m_mainPresenter` which are members of the test class. `Mock::VerifyAndClearExpectations` will be called automatically when the mock is destroyed but because it's a member that won't happen till all the tests have finished, so we have to call it manually at the end of each test. The newer tests create the mocks as local variables so they get destroyed at the end of each test, which is the better way to do it.

Reply:
Ah, thanks very much Gemma, that's really useful to know. So we should really see if we can refactor it to create the mocks as local variables when we look at the maintenance issue then.